docs/tutorial/getting-started.md

```diff
@@ -82,14 +112,63 @@ unit-jenkins-k8s-0:
     started: <timestamp>
 ```
 
-You can now access your Jenkins server UI at `http://<UNIT_IP>:8080` and login using username
-"admin" and password from the action above.
+You can now access your Jenkins server UI at the previous url, and login using username "admin" and password from the action above.
+
+### Deploy and integrate k8s agents
+
+By default, jenkins-k8s server application is installed with 0 executors for security purposes.
+A functional Jenkins application requires additional Jenkins agents to be integrated.
+
+The following commands deploy 3 units of the jenkins-agent-k8s charm and integrate them with the
+jenkins-k8s charm.
+
+```
+juju deploy jenkins-agent-k8s --channel=latest/edge --num-units=3
+
+# 'agent' relation name is required since jenkins-k8s charm provides multiple compatible
+# interfaces with jenkins-agent-k8s charm.
+juju integrate jenkins-k8s:agent jenkins-agent-k8s:agent
+```
+
+### Deploy and integrate machine agents (optional)
+
+For this section you need a machine model named `lxd`. 
+
+The first requirement is to create the offer, so the jenkins-k8s agent endpoint is available
+for cross-model integrations.
+
+```
+juju offer jenkins-k8s:agent
+```
+
+Once the offer is created, we can create the new model, deploy the machine jenkins-agent charm
+and integrate it with jenkins-k8s with:
+```
+juju add-model --controller=lxd jenkins-tutorial
+juju deploy --model lxd:jenkins-tutorial jenkins-agent --channel=latest/edge
+juju integrate --model lxd:jenkins-tutorial jenkins-agent:agent microk8s:admin/jenkins-tutorial.jenkins-k8s
+```
+
+You can check the status of the lxd model with:
+```
+juju status --model lxd:jenkins-tutorial --relations
+```
+
+After a few minutes you should be able to see the jenkins agent machine model as a new build executor
+in the Jenkins UI.
+
 
 ### Cleaning up the environment
 
 Congratulations! You have successfully finished the jennkins-k8s tutorial. You can now remove the
 model environment that you’ve created using the following command.
```

Comment:
Nitpick: We set up two different environments (assuming the user followed the optional part), so probably should say "You can now remove the model environments..."